pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -385,8 +387,7 @@ private BrokerResponseNative handleRequest(long requestId, String query,
       if (!hasTableAccess) {
         _brokerMetrics.addMeteredTableValue(tableName, BrokerMeter.REQUEST_DROPPED_DUE_TO_ACCESS_ERROR, 1);
         LOGGER.info("Access denied for request {}: {}, table: {}", requestId, query, tableName);
-        requestContext.setErrorCode(QueryException.ACCESS_DENIED_ERROR_CODE);
```

Comment:
We should still set the request context

Reply:
Please check all usage of `hasAccess()`, seems there are other places also need to be handled

Reply:
Checked and handled other calls of hasAccess.